File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProduct/IfcElementType/IfcElementComponentType/IfcReinforcingElementType/IfcReinforcingBarType/IfcReinforcingBarType.vb
Public Class IfcReinforcingBarType Inherits IfcReinforcingElementType
    Public PredefinedType As IfcReinforcingBarTypeEnum
    Public NominalDiameter As IfcPositiveLengthMeasure
    Public CrossSectionArea As IfcAreaMeasure
    Public BarLength As IfcPositiveLengthMeasure
    Public BarSurface As IfcReinforcingBarSurfaceEnum
    Public BendingShapeCode As IfcLabel
    Public BendingParameters As List(Of IfcBendingParameterSelect)

    ' === WHERE CLAUSES ===
    ' BendingShapeCodeProvided : NOT EXISTS(BendingParameters) OR EXISTS(BendingShapeCode)
    ' CorrectPredefinedType : (PredefinedType <> IfcReinforcingBarTypeEnum.USERDEFINED) OR ((PredefinedType = IfcReinforcingBarTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType))
End Class

File: src/main/resources/output/IFC4X3_java/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProduct/IfcElementType/IfcElementComponentType/IfcReinforcingElementType/IfcReinforcingBarType/IfcReinforcingBarType.java
public class IfcReinforcingBarType extends IfcReinforcingElementType {
    public IfcReinforcingBarTypeEnum PredefinedType;
    public IfcPositiveLengthMeasure NominalDiameter;
    public IfcAreaMeasure CrossSectionArea;
    public IfcPositiveLengthMeasure BarLength;
    public IfcReinforcingBarSurfaceEnum BarSurface;
    public IfcLabel BendingShapeCode;
    public List<IfcBendingParameterSelect> BendingParameters;

    // === WHERE CLAUSES ===
    // BendingShapeCodeProvided : NOT EXISTS(BendingParameters) OR EXISTS(BendingShapeCode)
    // CorrectPredefinedType : (PredefinedType <> IfcReinforcingBarTypeEnum.USERDEFINED) OR ((PredefinedType = IfcReinforcingBarTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType))
}
